src/librbd/cache/pwl/Types.h

```diff
@@ -295,8 +295,8 @@ struct WriteLogPoolRoot {
                                   * sync gen number are flushed. */
   uint32_t block_size;           /* block size */
   uint32_t num_log_entries;
-  uint32_t first_free_entry;     /* Entry following the newest valid entry */
-  uint32_t first_valid_entry;    /* Index of the oldest valid entry in the log */
+  uint64_t first_free_entry;     /* Entry following the newest valid entry */
+  uint64_t first_valid_entry;    /* Index of the oldest valid entry in the log */
```

Comment:
This is an on-disk format change.  I think we need to bump the layout version for `ssd` mode.

Reply:
Hrm, speaking of layout version, it's not even checked `ssd` mode.  Need to add that!

Reply:
> layout

I try to find `layout_version` in ssd mode. but, the author didn't use version control. cc @MahatiC

Reply:
> This is an on-disk format change. I think we need to bump the layout version for `ssd` mode.

I try to find layout_version in ssd mode. but, the author didn't use version control. cc @MahatiC

Reply:
Yes, layout_version isn't checked in ssd mode, but I wasn't involved in that part of design decision - so I'm not sure why. Will try to dig into docs for any info on this

Reply:
> Hrm, speaking of layout version, it's not even checked `ssd` mode. Need to add that!

Add layout version in this pr, or open a new PR to add in the future?

Reply:
In this PR -- we can't change the types without bumping the version and we can't bump the version without adding the check.

Reply:
> In this PR -- we can't change the types without bumping the version and we can't bump the version without adding the check. (Of course, the check should have been there from the start...)

I will open a separate PR for layout version. After version control pr merged,  change version here. Is this ok?

Reply:
I would prefer to do it here.  Currently the version is not only checked but also not set, so we can assume that version 0 stands for "broken".  The 64-bit log entry pointers would be version 1.  No need to create a separate PR for a few lines of code that are very closely related.